### PR TITLE
Feature/signup user: Init user DB & Create API for register user

### PR DESCRIPTION
**Pull request by ademrizki**

### CODE REVIEW

**Comment by ademrizki** [comment]
Screenshot 2024-10-23 at 14 00 22

File: routes/users.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"example.com/rest-api/models"
+	"github.com/gin-gonic/gin"
+)
+
+func signUp(context *gin.Context) {
+	var user models.User
+	err := context.ShouldBindJSON(&user)
+
+	if err != nil {
+		fmt.Println(err)
```
Comment (firzatullahd) [inline routes/users.go:16]: it is not a good idea to use fmt.Println for logging, it will not show the timestamp.

File: routes/users.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			Message:    err.Error(),
+		})
+		return
+	}
+
+	err = user.SaveUser()
+
+	if err != nil {
+		context.JSON(http.StatusInternalServerError, models.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
```
Comment (wlrudi19) [inline routes/users.go]: maybe you can sync with error code internal server error @ademrizki

**Review by wlrudi19 — APPROVED** [review]

File: db/db.go [hunk 3]
```diff
@@ -23,20 +23,36 @@ func InitDB() {
...
+		FOREIGN KEY(user_id) REFERENCES users(id)
 	)
 	`
 
-	_, err := DB.Exec(createEventTables)
+	_, err = DB.Exec(createEventTables)
 
 	if err != nil {
-		panic("Could not create table.")
+		panic("Could not create table event.")
```
Comment (firzatullahd) [inline db/db.go:56]: you should never use panic, return use logging, and return error instead.

**Pull request merged**